python/mxnet/executor_manager.py

```diff
@@ -283,8 +283,13 @@ def update_metric(self, metric, labels):
         """ Update evaluation metric with label and current outputs """
         for texec, islice in zip(self.train_execs, self.slices):
             n = int(texec.outputs[0].shape[0] / (islice.stop - islice.start))
-            new_slice = slice(islice.start * n, islice.stop * n)
-            labels_slice = [label[new_slice] for label in labels]
+            # special case for ctc loss
+            if len(labels) > 0 and len(labels[0].shape) == 2 \
+               and n > labels[0].shape[1]:
```

Comment:
I think the best way is let metric function itself to process these difference. So, I suggest revert this code back to previous implement.

`def update_metric(self, metric, labels):
        """ Update evaluation metric with label and current outputs """
        for texec, islice in zip(self.train_execs, self.slices):
            labels_slice = [label[islice] for label in labels]
            metric.update(labels_slice, texec.outputs)`